src/AdoNet/Orleans.Reminders.AdoNet/SiloHostBuilderReminderExtensions.cs

```diff
@@ -42,7 +42,7 @@ public static IServiceCollection UseAdoNetReminderService(this IServiceCollectio
             services.AddSingleton<IReminderTable, AdoNetReminderTable>();
             services.ConfigureFormatter<AdoNetReminderTableOptions>();
             services.AddSingleton<IConfigurationValidator, AdoNetReminderTableOptionsValidator>();
-            services.Configure(configureOptions);
+            configureOptions(services.AddOptions<AdoNetReminderTableOptions>());
```

Comment:
wow this is dangerous mistake which really easy to miss, since it doesn't have compile time validation. I wonder do we make the same mistake in our other part of code base..

Reply:
how do we find a way to solve the ambiguous between `Action<TOptions>` and `Action<OptionsBuilder<TOptions>>`. If we are making this mistake, our users would probably make the same mistake somewhere else.. @sergeybykov @BenjaminGibbs @jason-bragg @jdom

Reply:
The PR LGTM, but just want to bring up this issue.

Reply:
I suspect you meant @benjaminpetit, not @BenjaminGibbs? :-)

Reply:
yep :) thanks for pointing this out. And apologize to the other Ben. But welcome to leave feedback if you want :)

Reply:
I'm trying to find places where end-users would use the wrong signature too, but can'd think of any. It's not common to define a method that takes and OptionsBuilder, only infrastructure code like ours or 3rd party extensions would do this, to provide syntactic sugar for the end user. End users would just either use our extension methods or call `.Configure(Action<TOptions>)` and not use the builder itself.  Still, this is the API for Options, not Orleans, not much we should do if it was an issue (which I still don't believe it is).

Reply:
Yeah the only thing which bothered me as an user, is when both has defaults to null . For example : 
```
UseXXX(Action<TOptions> configure = null);
UseXXX(Action<OptionBuilder<TOptions>> configure = null)
```

As a user, if I want to call `UseXXX()` I will have to cast it somehow to determine which one. I definitly encounterred this before when writting tests, not sure if there's still occurence like this in our code base

Reply:
Yes, we should be consistent with our APIs - no ambiguities. Only the `Action<T>` overload should have a default value in my opinion. Someone can change those things in another PR

Reply:
yeah totally. I didn't intend to push those in this PR. it just triggered my thought. This pr is good.

Reply:
I can merge it if tests are passing in vso. @sergeybykov mentioned they weren't

Reply:
oh, sorry, just read this last comment. Nevertheless I did check that a newer VSO build was passing before merging

Reply:
I should have updated the thread to say that the tests were passing - my bad.